src/vivarium_gates_nutrition_optimization/components/hemoglobin.py

```diff
@@ -189,21 +206,53 @@ def sample_from_hemoglobin_distribution(
-    #     severe_ratio = probability["severe_maternal_hemorrhage"] / p_maternal_hemorrhage
-    #     p_maternal_hemorrhage_nonanemic = p_maternal_hemorrhage * (1 - paf)
-    #     p_maternal_hemorrhage_anemic = p_maternal_hemorrhage_nonanemic * rr
-    #     hemoglobin = self.hemoglobin(index)
-    #     anemic = hemoglobin <= SEVERE_ANEMIA_AMONG_PREGNANT_WOMEN_THRESHOLD
-    #     probability["severe_maternal_hemorrhage"] = severe_ratio * p_maternal_hemorrhage_nonanemic
-    #     probability["moderate_maternal_hemorrhage"] = (1 - severe_ratio) * p_maternal_hemorrhage_nonanemic
-    #     probability.loc[anemic, "severe_maternal_hemorrhage"] = severe_ratio * p_maternal_hemorrhage_anemic
-    #     probability.loc[anemic, "moderate_maternal_hemorrhage"] = (1 - severe_ratio) * p_maternal_hemorrhage_anemic
-    #     probability["not_maternal_hemorrhage"] = (1 - probability["moderate_maternal_hemorrhage"] - probability["severe_maternal_hemorrhage"])
-    #     return probability
+    def adjust_maternal_disorder_proportion(
+        self, index: pd.Index, maternal_disorder_probability: pd.DataFrame
+    ) -> pd.Series:
+        hemoglobin_level = self.hemoglobin(index)
+        rr = self.maternal_disorder_rr(index)
+        ## annoyingly formatted
+        paf = self.maternal_disorder_paf(index)["value"]
+        tmrel = TMREL_HEMOGLOBIN_ON_MATERNAL_DISORDERS
+        per_simulant_exposure = (
+            (tmrel - hemoglobin_level + abs(tmrel - hemoglobin_level)) / 2 / RR_SCALAR
+        )
+        per_simulant_rr = rr**per_simulant_exposure
+        maternal_disorder_probability *= (1 - paf) * per_simulant_rr
+        return maternal_disorder_probability.map(lambda value: 1 if value > 1 else value)
+
+    def adjust_maternal_hemorrhage_proportion(self, index, maternal_hemorrhage_probability):
+        paf = self.hemorrhage_paf(index)["value"]
+        rr = self.hemorrhage_rr(index)
+        hemoglobin = self.hemoglobin(index)
+        maternal_hemorrhage_probability *= 1 - paf
+        # Dichotomous risk based on severe anemia
+        maternal_hemorrhage_probability.loc[
+            hemoglobin <= SEVERE_ANEMIA_AMONG_PREGNANT_WOMEN_THRESHOLD
+        ] *= rr
+        return maternal_hemorrhage_probability
+
+    def adjust_hemoglobin_exposure(
+        self, index: pd.Index, hemoglobin_exposure: pd.DataFrame
+    ) -> pd.DataFrame:
+        pop = self.population_view.get(index)
+        # We need to persist this value for both current and recovered maternal hemorrhage
+        maternal_hemorrhage_mask = (pop["alive"] == "alive") & (
+            pop["maternal_hemorrhage"] != "susceptible_to_maternal_hemorrhage"
+        )
+        if not maternal_hemorrhage_mask.any():
+            return hemoglobin_exposure
+        # Choose hemorrhage severity and multiply hemoglobin exposure by appropriate hemorrhage scale factor
+        hemoglobin_exposure[maternal_hemorrhage_mask] *= self.randomness.choice(
```

Comment:
I'm not sure I fully understand what you're doing, but this seems wrong to me. If we perform this choice on every time-step, for a given simulant we'll get different random numbers on each time-step, and I believe we want this always to be the same for a given simulant. I think we want to draw once when they enter the hemorrhage state and persist that value rather than redrawing each time-step.

Reply:
agreed, fixed so that the sampling happens on initialization